bcipy/tasks/rsvp/stopping_criteria.py

```diff
@@ -0,0 +1,231 @@
+import numpy as np
+import logging
+from typing import Dict, List
+from copy import copy
+
+log = logging.getLogger(__name__)
+
+
+# Criteria
+class DecisionCriteria:
+    """Abstract class for Criteria which can be applied to evaluate a sequence
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    def reset(self):
+        pass
+
+    def decide(self, epoch):
+        """
+        Apply the given criteria.
+        Parameters:
+        -----------
+            epoch - Epoch data
+                - target(str): target of the epoch
+                - time_spent(ndarray[float]): |num_trials|x1
+                      time spent on the sequence
+                - list_sti(list[list[str]]): presented symbols in each
+                      sequence
+                - list_distribution(list[ndarray[float]]): list of |alp|x1
+                        arrays with prob. dist. over alp
+
+        """
+        raise NotImplementedError()
+
+
+class MinIterationsCriteria(DecisionCriteria):
+    """ Returns true if the minimum number of iterations have not yet
+        been reached. """
+
+    def __init__(self, min_num_seq):
```

Comment:
This is a good refactor. I agree that `min_num_seq` should be passed in as a constructor parameter rather than in the `decide` method. You could add a type annotation of `int` here.

Reply:
`int` was added by Aziz.